src/connector/src/base.rs

```diff
@@ -47,22 +52,93 @@ pub struct SourceMessage {
 }
 
 /// The metadata of a split.
-pub trait SourceSplit: Sized {
+pub trait SplitMetaData: Sized {
     fn id(&self) -> String;
-    fn to_string(&self) -> Result<String>;
+    fn to_json_bytes(&self) -> Bytes;
     fn restore_from_bytes(bytes: &[u8]) -> Result<Self>;
 }
 
 /// The persistent state of the connector.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ConnectorState {
     pub identifier: Bytes,
     pub start_offset: String,
     pub end_offset: String,
 }
 
+impl ConnectorState {
+    pub fn from_hashmap(state: HashMap<String, String>) -> Vec<Self> {
+        if state.is_empty() {
+            return vec![];
+        }
+        let mut connector_states: Vec<Self> = Vec::with_capacity(state.len());
```

Comment:
So actually ConnectorState means the state of one single split? Then why not call it SplitState? It makes me confuse to think of ConnectorState is the state of an entire connector.

Reply:
Also, SourceStateHandler should better be named SplitStateHandler or SplitStateStore

Reply:
let's unite the state-related stuff in another pr.

I plan to remove `ConnectorState` and use the connector-specific split description for state management.